futures-select-macro/Cargo.toml

```diff
@@ -22,4 +22,4 @@ std = []
 proc-macro2 = "0.4"
 proc-macro-hack = "0.5.3"
 quote = "0.6"
-syn = { version = "0.15.22", features = ["full"] }
+syn = { version = "0.15.25", features = ["full"] }
```

Comment:
Why was this change required?

Reply:
A warning occurred when building with minimal versions: https://travis-ci.com/rust-lang-nursery/futures-rs/jobs/209211855. (that warning was fixed in https://github.com/dtolnay/syn/pull/563)

(Context: After the first commit, I realized that [CI did not catch(and reject) the warnings](https://github.com/rust-lang-nursery/futures-rs/pull/1680#issuecomment-503437829), so I added eb4cbfa and changed the warnings to be denied for all builds.)